src/org/traccar/protocol/TotemProtocolEncoder.java

```diff
@@ -16,17 +16,25 @@
  */
 package org.traccar.protocol;
 
+import org.traccar.Context;
 import org.traccar.StringProtocolEncoder;
 import org.traccar.helper.Log;
 import org.traccar.model.Command;
+import org.traccar.model.Device;
 
 public class TotemProtocolEncoder extends StringProtocolEncoder {
 
     @Override
     protected Object encodeCommand(Command command) {
 
-        // Temporary put default password
         command.set(Command.KEY_DEVICE_PASSWORD, "000000");
+        if (Context.getDeviceManager() != null) {
```

Comment:
Do we really need all these null checks? I think if we get to the encoder stage, it means that device definitely exists.

Also, you shouldn't call `toString()` method. Just cast result to string.